### DISCUSSION TITLE
Unit testing

### DISCUSSION THREAD

**Discussion by ricoderks** [ndcn/omicser#181] [Ideas]
@ergonyc maybe a good idea to start work on the unit testing in a separate branch. Probably easiest to start with creating tests for the smaller functions. For testing graphical output we maybe can try to use [`vdiffr` package](https://vdiffr.r-lib.org/).

**Comment by ergonyc** [comment]
Sounds good.  Please go ahead and make a dev-testing branch.  I think we want to quickly roll this into the main / main-dev workflow.  With the v2.0 (dataset linking) release?

steps:

- [ ] testing branch to impliment
- [ ] proof of concept (working testing branch)
- [ ] initial code coverage: data imported, visualizations return values, etc.
- [ ] integration into main/main-dev branch
- [ ] goals for code coverage detailed in roadmap

**Comment by ricoderks** [comment]
I set up a dev-testing branch in my own repo. Made a small test with testthat and tried to get codecoverage calculated. I succeeded to get codecoverage calculated, but for some reason I can't get it into the READ.Rmd file. I think it has to do with the branch name, but I tried already serveral things. No luck so far.
I also noticed that the github actions are not running anymore. They fail on installing the package complexheatmap from github as source. Looks like the circlize package is the problem. I'll look into this a bit more.

**Reply by ergonyc** [reply]
@ricoderks good job nailing down the circlize regression.   I think we should start integrating your dev-testing with our main-dev branch... and maybe add some boilerplate guidelines around testing?

**Reply by ddifraia** [reply]
@ricoderks just read this now, but this could be related to the fact that complexheatmap and circlize (? not sure about this one) are part of biocoductor repository and not CRAN. To overcome this problem you should add to the DESCRIPTION file the entry biocViews: with no attributes. Should work in theory.

**Reply by ricoderks** [reply]
@ddifraia This is fixed. The developer of complexheatmap is also the deloper of circlize. He update the CRAN version of circlize.
